### PR TITLE
Respect allowaccountssameemail setting

### PR DESCRIPTION
**Pull request by dmitriim**

### CODE REVIEW

File: auth.php [hunk 1]
```diff
@@ -697,6 +715,57 @@ public function update_user_profile_fields(&$user, $attributes, $newuser = false
...
 
+    /**
+     * Get email address from attributes.
+     *
+     * @param array $attributes A list of attributes.
+     *
+     * @return bool
+     */
+    public function get_email_from_attributes(array $attributes) {
+        $emailattr = get_config('auth_saml2', 'field_map_email');
```
Comment (brendanheywood) [inline auth.php]: why not use $this->config ?

Comment (dmitriim) [reply auth.php]: I have copied it from https://github.com/catalyst/moodle-auth_saml2/blob/master/auth.php#L659 But you are right we should user $this->config here. Will fix it.

Comment (dmitriim) [reply auth.php]: Refactored the code to use $this->config

**Comment by brendanheywood** [comment]
This is good to go,just waiting for travis

**Pull request merged**